storage/storage.go

```diff
@@ -1132,6 +1135,39 @@ func (o *ObjectAttrs) toRawObject(bucket string) *raw.Object {
 	}
 }
 
+// toProtoObject copies the editable attributes from o to the proto library's Object type.
+func (o *ObjectAttrs) toProtoObject(b string) *storagepb.Object {
+	var ret *timestamppb.Timestamp
+	if !o.RetentionExpirationTime.IsZero() {
+		ret = timestamppb.New(o.RetentionExpirationTime)
+	}
+	var ct *timestamppb.Timestamp
+	if !o.CustomTime.IsZero() {
+		ct = timestamppb.New(o.CustomTime)
+	}
+	// For now, there are only globally unique buckets, and "_" is the alias
+	// project ID for such buckets.
+	b = bucket("_", b)
+
+	return &storagepb.Object{
+		Bucket:              b,
+		Name:                o.Name,
+		EventBasedHold:      proto.Bool(o.EventBasedHold),
+		TemporaryHold:       o.TemporaryHold,
+		RetentionExpireTime: ret,
+		ContentType:         o.ContentType,
+		ContentEncoding:     o.ContentEncoding,
+		ContentLanguage:     o.ContentLanguage,
+		CacheControl:        o.CacheControl,
+		ContentDisposition:  o.ContentDisposition,
+		StorageClass:        o.StorageClass,
+		Acl:                 toProtoObjectACL(o.ACL),
+		Metadata:            o.Metadata,
+		CustomTime:          ct,
+		KmsKey:              o.KMSKeyName,
```

Comment:
Why is it necessary to add this field here and not in the apiary object?

Reply:
Idk if it's necessary or not, but it's in the proto so I included it here. Should I remove it?

Reply:
Hmm, looks like it might be an oversight in the current implementation which is concerning!